### PR TITLE
Add Makefile, use new .pre-commit-config.yaml

### PR DESCRIPTION
**Pull request by mikapfl**
Closes: #134 

* Adds a Makefile with commonly used commands in short form
* Also properly adds the .pre-commit-config.yaml so that it will be distributed asap.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+		$(MAKEFILE_LIST) | column -ts '^'
+
+build:          ## Build the package using lucode2::buildLibrary()
+	Rscript -e 'lucode2::buildLibrary()'
+
+check:          ## Build documentation and vignettes, run testthat tests,
+                ## and check if code etiquette is followed using lucode2::check()
+	Rscript -e 'lucode2::check()'
+
+test:           ## Run testthat tests
```
Comment (orichters) [inline Makefile:15]: Minor detail: maybe sort those alphabetically

**Review by orichters — APPROVED** [review]
great, much appreciated!

File: Makefile (hunk 1 above)
Comment (mikapfl) [reply Makefile:15]: I sorted them by "what you use most often, according to my gut feeling".

Comment (orichters) [reply Makefile:15]: fine as well…

**Review by pfuehrlich-pik — APPROVED** [review]

**Comment by pfuehrlich-pik** [comment]
Cool stuff, thanks!

**Comment by pfuehrlich-pik** [comment]
Do you know how easy/hard it is to get `make` running on Windows?

**Comment by mikapfl** [comment]
> Do you know how easy/hard it is to get `make` running on Windows?

Never tried it, I guess it is possible via gnuwin or WSL?

**Pull request merged**